pgmpy/estimators/ConstraintBasedEstimator.py

```diff
@@ -0,0 +1,648 @@
+        ... model = BayesianModel([('A', 'C'), ('B', 'C'), ('B', 'D'), ('C', 'E')])
+        >>> skel, sep_sets = ConstraintBasedEstimator.build_skeleton(model.nodes(), model.get_independencies())
+        >>> print(skel.edges())
+        [('A', 'C'), ('B', 'C'), ('B', 'D'), ('C', 'E')]
+        """
+
+        nodes = list(nodes)
+
+        if isinstance(independencies, Independencies):
+            def is_independent(X, Y, Zs):
+                return IndependenceAssertion(X, Y, Zs) in independencies
+        elif callable(independencies):
+            is_independent = independencies
+        else:
+            raise ValueError("'independencies' must be either Independencies-instance " +
+                             "or a ternary function that decides independencies.")
+
+        graph = UndirectedGraph(combinations(nodes, 2))
+        lim_neighbors = 0
+        seperating_sets = dict()
+        while not all([len(graph.neighbors(node)) < lim_neighbors for node in nodes]):
+            for node in nodes:
+                for neighbor in graph.neighbors(node):
+                    # search if there is a set of neighbors (of size lim_neighbors)
+                    # that makes X and Y independent:
+                    for seperating_set in combinations(set(graph.neighbors(node)) - set([neighbor]), lim_neighbors):
+                        if is_independent(node, neighbor, seperating_set):
+                            seperating_sets[frozenset((node, neighbor))] = seperating_set
+                            graph.remove_edge(node, neighbor)
+                            break
+            lim_neighbors += 1
+
+        return graph, seperating_sets
+
+    def test_conditional_independence(self, X, Y, Zs=[]):
+        """Chi-square conditional independence test.
+        Tests if X is independent from Y given Zs in the data.
+
+        This is done by comparing the observed frequencies with the expected
+        frequencies if X,Y were conditionally independent, using a chisquare
+        deviance statistic. The expected frequencies given independence are
+        `P(X,Y,Zs) = P(X|Zs)*P(Y|Zs)*P(Zs)`. The latter term can be computed
+        as `P(X,Zs)*P(Y,Zs)/P(Zs).
+
+        Parameters
+        ----------
+        X: int, string, hashable object
+            A variable name contained in the data set
+        Y: int, string, hashable object
+            A variable name contained in the data set, different from X
+        Zs: list of variable names
+            A list of variable names contained in the data set, different from X and Y.
+            This is the seperating set that (potentially) makes X and Y independent.
+            Default: []
+
+        Returns
+        -------
+        p_value: float
+            A significance level for the hypothesis that X and Y are dependent
+            given Zs. The p_value is the probability of falsely rejecting the
+            hypothesis that the variables are conditionally dependent. A low
```

Comment:
Fixed, this was confused. I also changed the return type, now the docstring reads:

```
        Returns
        -------
        chi2: float
            The chi2 test statistic.
        p_value: float
            The p_value, i.e. the probability of observing the computed chi2
            statistic (or an even higher value), given the null hypothesis
            that X _|_ Y | Zs.
        sufficient_data: bool
            A flag that indicates if the sample size is considered sufficient.
            As in [4], require at least 5 samples per parameter (on average).
            That is, the size of the data set must be greater than
            `5 * (c(X) - 1) * (c(Y) - 1) * prod([c(Z) for Z in Zs])`
            (c() denotes the variable cardinality).

```